### ISSUE TITLE
Flow says function is missing, when actually it's just called incorrectly

### ISSUE THREAD

**Issue by pnarielwala-tc** [facebook/flow#7828]
Hello!

It seems like there is a bug/change in behavior when a function is called with invalid arguments. Currently flow error says that the function cannot be found in the props object when in fact it IS there, just called incorrectly.

[TryFlow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAlgWwA5wE4AuYASgKYCGAxgQDRgDeYAwnDnAHantEC+YUeVmADkeCtWGpUBAJ7ZSYAPIJ2ABUHYAzgBUwAXgaowYeJXIF0HAFxhNBPOnYBzGqh7S5CsgBMArgA91OC1dA3ojMAAZOCdHG0UAIwArUmpXd1l5MAARdE1sc0oACyCQ-UNjTV8EzHQCGwAKb3NyG3pKOG9SGzsHZx4ASn0APjAANzh0b1djYGBbatr6sCaWtsGR8cnpsDmwBCLzfngkPNsKPGLHJ358MEc7Cm83KUyFUp1y5TUNT4AyEikPyBX66AG5fKFEqgqSUGDkTSaMAAWRkLDYnG4YFI-gIXG8SPRuExBAAPB9tMNwsYOhiuARct5kXBfNwGkNqTSOHYGCY4GYLBw6FUanUwHwDAQinkAHTYX4RBaiggNegbPakPCCPC2IosmDeMAJBRFTWkOhiACOvnQYkN5DwTl8mHpSIdCnYcCIBURQIi7h4QA)
In the example above we get a different error than what we would get in `v0.75`
```
// v0.101.0
all branches are incompatible: Either property `submit` is missing in `OwnPropsT` [1]. Or property `submit` is missing in `ReduxPropsT` [2].

// v0.75.0
Cannot call `submit` with object literal bound to `data` because property `code` is missing in object literal [1] but exists in object type [2].
```

[TryFlow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgAKATnDgM4AqYAvGAN6phhRxwBcYAFACYCGGPp3oBjODzydyGYgEsAdgHMAvgEpaAPjAA3OLJ4AaVMvRj50sAFssAMQCu8kRllx5tbjlIVOJMlXU0WowAkGYW9Kxwyu6efkzMLGxc9GrGQA)
In other example, there is no error in the latest version but we get an (correct) error in `v0.75`
```
// v0.101.0
No errors!

// v0.75.0
Cannot call `foo` with object literal bound to `data` because property `code` is missing in object literal [1] but exists in object type [2].
```

Thanks!

**Comment by dsainati1** [comment]
In your second example, this is the expected behavior because of how Flow treats unsealed objects. See #7345.

The first example appears to be a poorly positioned error due to destructuring, which I believe we are working on fixing.

**Comment by pnarielwala-tc** [comment]
So for the second example, if I make the parameter object an exact type but the property inside optional, should I be getting an error?

[TryFlow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgAKATnDgM4AqYAvGAN6phhRxwBcYAFACYCGGPp3oAfAMZweeAPydyGYgEsAdgHMRAXwCUtAHxgAbnEU8ANKg3oJy+WAC2WAGIBXZWIyK4y2txykKnCRkVDo0+owAkNa29KxwGj5+wUzMLGxc9NoWQA)

**Comment by dsainati1** [comment]
Yes, because `{}` is an inexact object.

**Comment by pnarielwala-tc** [comment]
Hmm even if `code` is "optional"? Or is flow doing away with optional object properties?

[TryFlow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgAiAhhsQCpgC8YA3gD4DGcAJngPwBcYAzhgE4BLAHYBzegF9UzYXzAtSxbiTKUatCWGDAwefvzj92YALRgARgFcMYZmzCCeYAERwcGQXGHEYz6V7kFMgAmZUU1Ojs8bgByGM1tMGE4XX1DIA)

**Comment by dsainati1** [comment]
Exactness and inexactness are separate concepts from optional properties. An inexact object can _never_ be assigned to an exact one, even if they have the same properties: https://flow.org/try/#0DYUwLgBAHhBcEG8CGcIGYC+EC8iX0wG4AoY0SAT1QQB990atdlUjiIPPPircpCugoA

**Comment by pnarielwala-tc** [comment]
I understand the exact and inexact concepts. My main question is how do [`optional` properties](https://flow.org/en/docs/types/objects/#toc-optional-object-type-properties) fit in with exact types? Or do they not?

https://github.com/facebook/flow/issues/2386

**Comment by SamChou19815** [comment]
Fixed in 0.191.0
